tests/test_views.py

```diff
@@ -492,3 +493,46 @@ def test_refresh_jwt_after_refresh_expiration(self):
     def tearDown(self):
         # Restore original settings
         api_settings.JWT_ALLOW_REFRESH = DEFAULTS['JWT_ALLOW_REFRESH']
+
+
+class DeviceViewTests(TokenTestCase):
+    def setUp(self):
+        super(DeviceViewTests, self).setUp()
+        self.device = Device.objects.create(
+            user=self.user, jwt_secret='somestring', permanent_token='somestring2', name='Android',
+            last_request_datetime=datetime.now())
+        self.user2 = User.objects.create_user(email='jsmith@example.com', username='jsmith', password='password')
+        self.device2 = Device.objects.create(
+            user=self.user2, jwt_secret='somestring99', permanent_token='somestring98', name='Android',
+            last_request_datetime=datetime.now())
+
+    def _login(self, client):
+        client.credentials(HTTP_AUTHORIZATION='JWT {}'.format(self.get_token()))
+        return client.login(**self.data)
+
+    def test_device_delete(self):
+        client = APIClient(enforce_csrf_checks=True)
+        # test accessing without being logged in
+        response = client.delete('/devices/{}/'.format(self.device.permanent_token))
+        self.assertEqual(response.status_code, 401)
+
+        self._login(client)
+        # try removing device linked to other user
+        response = client.delete('/devices/{}/'.format(self.device2.permanent_token))
+        self.assertEqual(response.status_code, 404)
+        # test regular case
+        self.assertEqual(Device.objects.filter(permanent_token=self.device.permanent_token).count(), 1)
+        response = client.delete('/devices/{}/'.format(self.device.permanent_token))
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(Device.objects.filter(permanent_token=self.device.permanent_token).count(), 0)
+
+    def test_device_list(self):
+        client = APIClient(enforce_csrf_checks=True)
+        self._login(client)
+        response = client.get('/devices/', format='json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(set(response.data[0].keys()), {
+            'permanent_token', 'jwt_secret', 'created', 'name', 'details', 'last_request_datetime'
```

Comment:
Please, do not make any field *_secret_* public. After all, there is a reason why they call it secret.
http://getblimp.github.io/django-rest-framework-jwt/#jwt_secret_key